### PR TITLE
feat: add capacitor completion spec

### PR DESCRIPTION
**Pull request by tmwrnr**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
n/a

**What is the new behavior (if this is a feature change)?**
Adds a new spec for [capacitor](https://capacitorjs.com/docs/cli)

**Additional info:**
I also added a `cap.ts` spec file because the capacitor CLI can be used either with `capacitor` or `cap` ([Reference](https://github.com/ionic-team/capacitor/blob/22c2fb505ee80dfce335c05f841ffafd21b7842c/cli/package.json#L34)). I was not sure if this is ok or if there is a better way of doing this. If you prefer not to have the second spec I can remove it again.

### CODE REVIEW

File: src/capacitor.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+const platforms: Fig.Suggestion[] = [
+  { name: "android", icon: "fig://icon?type=android" },
+  { name: "ios", icon: "fig://icon?type=apple" },
+];
+
+const helpOption: Fig.Option = {
+  name: ["--help", "-h"],
+  description:
+    "Output usage information. Can be used with individual commands too",
+};
```
Comment (fedeci) [inline src/capacitor.ts:10]: If you add this property the option become available globally to all subcommands so it no longer needs to be repeated and can be moved directly in the completion object
```suggestion
  name: ["--help", "-h"],
  description:
    "Output usage information. Can be used with individual commands too",
  isPersistent: true,
};
```

File: src/capacitor.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        {
+          name: "--list",
+          description:
+            "Print a list of target devices available to the given platform",
+        },
+        {
+          name: "--target",
+          description: "Run on a specific target device",
+          args: {
+            name: "target",
```
Comment (fedeci) [inline src/capacitor.ts:80]: Can you add a [generator](https://fig.io/docs/reference/generator) here? Ping me if you need help!

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks! It seems capacitor can run through yarn and npx, could you add it to those specs too?

File: src/capacitor.ts (hunk 1 above)
Comment (tmwrnr) [reply src/capacitor.ts]: Thanks that is way better 👍

File: src/capacitor.ts [hunk 3]
```diff
@@ -3,10 +3,37 @@ const platforms: Fig.Suggestion[] = [
...
+      .slice(2) // Ignore output header lines
+      .map((s) => {
+        const [name, api, targetId] = s.replace(/\s\s+/g, "|").split("|");
+        return {
+          name: targetId,
+          displayName: `${name} ${api}`,
+          icon: "📱",
+        };
+      });
+  },
```
Comment (fedeci) [inline src/capacitor.ts:36]: Love it!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**